src/sbml/SBMLDocument.cpp

```diff
@@ -278,25 +278,19 @@ SBMLDocument::SBMLDocument (const SBMLDocument& orig)
 {
   
   
-  setSBMLDocument(this);
+  SBMLDocument::setSBMLDocument(this);
   
   mInternalValidator->setDocument(this);
   mInternalValidator->setApplicableValidators(orig.getApplicableValidators());
   mInternalValidator->setConversionValidators(orig.getConversionValidators());
   
   if (orig.mModel != NULL) 
   {
-    mModel = static_cast<Model*>( orig.mModel->clone() );
+    mModel = orig.mModel->clone();
```

Comment:
With model the static_cast is necessary because in the comp package a ModelDefinition derives from Model

Reply:
The cast can be done implicitly in this case, since `mModel` is defined as a `Model*`.